src/compiler/commandLineParser.ts

```diff
@@ -1449,7 +1461,7 @@ namespace ts {
             case "number":
                 return mapDefined(values, v => validateJsonOptionValue(opt.element, parseInt(v), errors));
             case "string":
-                return mapDefined(values, v => validateJsonOptionValue(opt.element, v || "", errors));
+                return mapDefined(values, v => validateJsonOptionValue(opt.element, opt.listPreserveFalsyValues ? v : (v || ""), errors));

```

Comment:
I see why `listPreserveFalsyValues` was needed for the `filter`s below, but I don’t understand why it was needed in this `map`.

Reply:
You're right, it's not needed. I removed it.